Fix AHBot SetPricesOfItem

Elmsroth opened the pull request:
Fix a bug in SetPricesOfItem method - Make server crash with AHBot enabled in debug mode.
![err1](https://user-images.githubusercontent.com/1208311/81448208-4d7a7800-917e-11ea-9aa3-3be9a010ea23.png)

![err2](https://user-images.githubusercontent.com/1208311/81448241-5cf9c100-917e-11ea-8c40-1a90dcd05fd5.png)

leprasmurf commented on src/shared/Utilities/Util.cpp:
```diff
@@ -736,3 +736,16 @@ void print_banner()
+// Sometimes needed whend using urand()
+void InvertMinMaxIfNeeded(uint32& min, uint32& max)
+{
+    uint32 minCopy = min;
+    uint32 maxCopy = max;
+
+    if (max < min)
+    {
+        min = maxCopy;
+        max = minCopy;
+    }
```
This can be accomplished with one fewer copy var.

copy = min;
if...
  min = max;
  max = copy;
}

Elmsroth replied:
yeah you are right

leprasmurf commented on src/game/AuctionHouseBot/AuctionHouseBot.cpp:
```diff
@@ -1809,11 +1809,18 @@ void AuctionBotSeller::SetPricesOfItem(AHB_Seller_Config& config, uint32& buyp,
-    double randrange = temp_buyp * 0.4;
-    buyp = (urand(temp_buyp - randrange, temp_buyp + randrange) / 100) + 1;
+    double randrange = temp_buyp * 0.4;    
+    uint32 buypMin = temp_buyp - randrange;
+    uint32 buypMax = temp_buyp + randrange;
+    InvertMinMaxIfNeeded(buypMin, buypMax);
+    buyp = (urand(buypMin, buypMax) / 100) + 1;
+    
     double urandrange = buyp * 40;
     double temp_bidp = buyp * 50;
-    bidp = (urand(temp_bidp - urandrange, temp_bidp + urandrange) / 100) + 1;
+    uint32 bidPmin = temp_bidp - urandrange;
+    uint32 bidPmax = temp_bidp + urandrange;
+    InvertMinMaxIfNeeded(bidPmin, bidPmax);
+    bidp = (urand(bidPmin, bidPmax) / 100) + 1;
```
This seems like a lot of variables to define.  My understanding of the issue is that sometimes randrange is negative and results in bad values being fed into `urand`.  If that's the case could the bug be mitigating by using the absolute value of `randrange`?

leprasmurf replied:
perhaps an explicit casting for the urand method and a modulus on the uint32 max for the other issues you mentioned in discord.

Elmsroth replied:
@leprasmurf , we have discussed about it with @billy1arm 
Better will be to check if overflow is detetected  and cap value to max uint32

Elmsroth replied:
@leprasmurf @billy1arm 
The PR is cleaned-up

Elmsroth committed:
Fix AHBot SetPricesOfItem

Fix a bug in SetPricesOfItem method - Make server crash with AHBOt enabled in debug mode
